students/ScottL/lesson10/mailroom_functional.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env python3
+
+# -------------------------------------------------#
+# Title: mailroom functional programming assignment
+# Dev: Scott Luse
+# Date: Mar 22, 2018
+# -------------------------------------------------#
+
+
+# -- Data --#
+donor_dict = {
```

Comment:
The fact that you have this global variable out here means it can be messed with and defeats the idea behind functional.  Having this global also "saves state" inside of this .py file which we also want to avoid.  A better move is have the dict create a base dict upon launching or have it input the base data from a file and then pass the dict around your functions rather than referencing global variables like this.